apps/webpack.config.js

```diff
@@ -75,6 +84,46 @@ const nodePolyfillConfig = {
       process: 'process/browser',
       timers: 'timers-browserify',
     }),
+    new CircularDependencyPlugin({
+      // ignore everything in a build directory or mode_modules
+      exclude: /node_modules|build/,
+      failOnError: true,
+      allowAsyncCycles: false,
+      cwd: process.cwd(),
+      // when we start, we re-initialize our list of previously seen circles to whatever
+      // we loaded from circular_depencies.json. If that file changes and you need to update, restart
+      // webpack
+      onStart: () => {
+        seenCircles.clear();
+        seenCircles = new Set(Array.from(circleSet));
+      },
+      onDetected: ({module: webpackModuleRecord, paths, compilation}) => {
+        const pathString = paths.join(' -> ');
+        // if the path is not a known existing one, then note as an error
+        if (!circleSet.has(pathString)) {
+          compilation.errors.push(
+            new Error(
+              `Circular Dependency Checker : New Circular Dependency found : ${pathString}`
+            )
+          );
+        }
+        // and since we've seen that path, we can delete it from our set of seen values
+        seenCircles.delete(pathString);
+      },
+      // finally, at the end, if we still have any circles that we previously knew about but did not see
+      // this time, note it as a warning.
+      onEnd: ({compilation}) => {
+        if (seenCircles.size > 0) {
+          compilation.warnings.push(
+            new Error(
+              `Circular Dependency Checker : Resolved circular dependencies can be removed from circles.json : ${Array.from(
```

Comment:
nit; 
```suggestion
              `Circular Dependency Checker : Resolved circular dependencies can be removed from circular_dependencies.json : ${Array.from(
```

Also, were you going to update the variable names from `circles` to `circularDependencies` as well? I do think that would improve consistency

Reply:
I just forgot. 🤭